evm/src/fuzz/invariant/filters.rs

```diff
@@ -0,0 +1,127 @@
+                    (excluded.is_empty() || !excluded.contains(addr))
+            })
+            .map(|(addr, (name, abi))| (addr, (name, abi, vec![])))
+            .collect();
+
+        self.select_selectors(invariant_address, abi, &mut contracts)?;
+
+        Ok((senders, contracts))
+    }
+
+    /// Selects the functions to fuzz based on the contract method `targetSelectors() -> (address,
+    /// bytes4[])[]`.
+    pub fn select_selectors(
+        &self,
+        address: Address,
+        abi: &Abi,
+        targeted_contracts: &mut TargetedContracts,
+    ) -> eyre::Result<()> {
+        let selectors =
+            self.get_list::<(Address, Vec<FixedBytes>)>(address, abi, "targetSelectors");
+
+        fn add_function(
+            name: &str,
+            selector: FixedBytes,
+            abi: &Abi,
+            funcs: &mut Vec<Function>,
+        ) -> eyre::Result<()> {
+            let func = abi
+                .functions()
+                .into_iter()
+                .find(|func| func.short_signature().as_slice() == selector.as_slice())
+                .wrap_err(format!("{name} does not have the selector {:?}", selector))?;
+
+            funcs.push(func.clone());
+            Ok(())
+        }
+
+        for (address, bytes4_array) in selectors.into_iter() {
+            if let Some((name, abi, address_selectors)) = targeted_contracts.get_mut(&address) {
+                // The contract is already part of our filter, and all we do is specify that we're
+                // only looking at specific functions coming from `bytes4_array`.
+                for selector in bytes4_array {
+                    add_function(name, selector, abi, address_selectors)?;
+                }
+            } else {
+                let (name, abi) = self.setup_contracts.get(&address).wrap_err(format!(
+                    "[targetSelectors] address does not have an associated contract: {}",
+                    address
+                ))?;
+                let mut functions = vec![];
+                for selector in bytes4_array {
+                    add_function(name, selector, abi, &mut functions)?;
+                }
+
+                targeted_contracts.insert(address, (name.to_string(), abi.clone(), functions));
+            }
+        }
+        Ok(())
+    }
+
+    /// Gets list of `T` by calling the contract `method_name` function.
+    fn get_list<T>(&self, address: Address, abi: &Abi, method_name: &str) -> Vec<T>
+    where
+        T: Tokenizable + Detokenize + TokenizableItem,
+    {
+        let mut list: Vec<T> = vec![];
+
+        if let Some(func) = abi.functions().into_iter().find(|func| func.name == method_name) {
+            if let Ok(call_result) = self.executor.call::<Vec<T>, _, _>(
+                CALLER,
+                address,
+                func.clone(),
+                (),
+                U256::zero(),
+                Some(abi),
+            ) {
+                list = call_result.result;
```

Comment:
return here directly so no need for the `list` var
instead you can return `Vec::new()` at the end of the function